chalice/deploy/deployer.py

```diff
@@ -596,16 +596,29 @@ def _get_lambda_memory_size(self, config):
             return DEFAULT_LAMBDA_MEMORY_SIZE
         return config.lambda_memory_size
 
+    def _can_reuse_deployment_package(self, config, lambda_name,
+                                      deployment_package_filename):
+        # type: (Config, str, str) -> bool
+        if not self._osutils.file_exists(deployment_package_filename):
+            return False
+        lambda_config = self._aws_client.get_function_configuration(
```

Comment:
It looked like it would require too much refactoring of things not related to my PR so I didn't but yes that call is made twice which is inefficient.